chore: Update README with backward compat. information (#598)
supabase/gotrue: README.md

3. `docker ps` should show 2 docker containers (`gotrue_postgresql` and `gotrue_gotrue`)

> 4. That's it! Visit the [health checkendpoint](http://localhost:9999/health) to confirm that gotrue is running.

Reviewer: /nit know this isn't in the PR but should this be 
```suggestion
4. That's it! Visit the [health check endpoint](http://localhost:9999/health) to confirm that gotrue is running.
```

## Running in production

Running an authentication server in production is not an easy feat. We

> recommend using [Supabase Auth](https://supabase.com/auth) which gets regular

Reviewer: Supabase Auth 🎉

security updates.

Otherwise, please make sure you setup a process to promptly update to the
latest version. You can do that by following this repository, specifically the
[Releases](https://github.com/supabase/gotrue/releases) and [Security
Advisories](https://github.com/supabase/gotrue/security/advisories) sections.

You should also subscribe to the
[supabase-gotrue-announcements](https://groups.google.com/g/supabase-gotrue-announcements)
Google Group to get email updates for important releases.

### Backward compatibility

GoTrue uses the [Semantic Versioning](https://semver.org) scheme. Here are some
further clarifications on backward compatibility guarantees:

**Go API compatibility**

GoTrue is not meant to be used as a Go library. There are no guarantees on
backward API compatibility when used this way regardless which version number
changes.

**Patch**

Changes to the patch version guarantees backward compatibility with:

- Database objects (tables, columns, indexes, functions).
- REST API
- JWT structure

> - Configuration

Reviewer: Should we specify that this refers to the names of the configuration values which live in the [Environment Variable files](https://github.com/supabase/gotrue/blob/master/example.env)?

Imagine that we might want to change the properties of the logger down the road which are also part of the configuration atm

Not guaranteed examples:

- Removal of JWT fields after a deprecation notice.
- Removal of certain APIs after a deprecation notice.
- Removal of sign-in with external providers, after a deprecation notice.
- Deletion, truncation, significant schema changes to tables, indexes, views,
  functions.

We aim to provide a deprecation notice in execution logs for at least two major
version releases or two weeks if multiple releases go out. Compatibility will
be guaranteed while the notice is live.

**Major**

Changes to the major version do not guarantee any backward compatibility with
previous versions.

### Inherited features

Certain inherited features from the Netlify codebase are not supported by
Supabase and they may be removed without prior notice in the future. This is a
comprehensive list of those features:

1. Multi-tenancy via the `instances` table i.e. `GOTRUE_MULTI_INSTANCE_MODE`
   configuration parameter.
2. System user (zero UUID user).
3. Super admin via the `is_super_admin` column.
4. SAML sign-in provider via the `GOTRUE_SAML_ENABLED` configuration
   parameter. (A different implementation for SAML may appear in the future
   which will be supported.)

> 5. Support for MySQL based databases. (Only Postgres is supported.)

Reviewer: Hmm are we using any Postgres specific features which would completely block a user from self-hosting on MySQL?

If not we could consider stating that it's possible but not natively supported (e.g. we won't have the starter files and we won't be taking queries about self-hosting with MySQL but the dedicated user could self-host with MySQL if they added additional code). 

This defers the decision of tying GoTrue to Postgres and allows us to eventually come back and support other Databases if we want to. At the same time, I realise it is quite unlikely that we will want to support other DBs given that the rest of Supabase tooling is heavily dependent on Postgres.

```suggestion
5. Built-in support for MySQL based databases. (We primarily support Postgres)
```

Reviewer 2: imo @hf 's wording is sane. If we want to add support for other DBs in the future that's fine, we can always come back and update this.

Reviewer: Ok! Resolving

3. `docker ps` should show 2 docker containers (`gotrue_postgresql` and `gotrue_gotrue`)
4. That's it! Visit the [health checkendpoint](http://localhost:9999/health) to confirm that gotrue is running.

## Running in production

Running an authentication server in production is not an easy feat. We
recommend using [Supabase Auth](https://supabase.com/auth) which gets regular
security updates.

If you still insist on running it on your own, please make sure you setup a
process to promptly update to the latest version. You can do that by following
this repository, specifically the
[Releases](https://github.com/supabase/gotrue/releases) and [Security
Advisories](https://github.com/supabase/gotrue/security/advisories) sections.

You should also subscribe to the

> [supabase-gotrue-announcements](https://groups.google.com/g/supabase-gotrue-users)

Author: Need to create the group before merging, ofc.
